src/utils/componentSwapData.ts

```diff
@@ -153,75 +161,36 @@ export async function getIssuanceComponentSwapData(
+      //   buyUnderlyingAmount,
+      //   // TODO: get swap data dynamically using 0x (restricted to Uni v3)
+      //   // TODO: construct swap data request from input/output token and buy amount
+      //   dexData: getStaticIssuanceSwapData(inputToken, underlyingERC20.address),
+      // }
+    })
+  const swapDataResults = await Promise.all(swaps)
   const swapData = issuanceComponents.map((_: string, index: number) => {
     const wrappedToken = wrappedTokens[index]
     const underlyingERC20 = wrappedToken.underlyingErc20
     const buyUnderlyingAmount = buyAmounts[index]
+    const dexData = swapDataResults[index]?.swapData ?? emptySwapData
     return {
       underlyingERC20: underlyingERC20.address,
       buyUnderlyingAmount,
-      dexData: getStaticIssuanceSwapData(inputToken, underlyingERC20.address),
-    }
-  })
-  return swapData
-}
-
-export async function getIssuanceERC4626SwapData(
-  indexTokenSymbol: string,
-  indexToken: string,
-  inputToken: string,
-  indexTokenAmount: BigNumber,
-  provider: JsonRpcProvider
-): Promise<ComponentSwapData[]> {
-  const issuanceModule = getIssuanceModule(indexTokenSymbol)
-  const issuance = new Contract(issuanceModule.address, IssuanceAbi, provider)
-  const [issuanceComponents] = await issuance.getRequiredComponentIssuanceUnits(
-    indexToken,
-    indexTokenAmount
-  )
-  const underlyingERC20sPromises: Promise<WrappedToken>[] =
-    issuanceComponents.map((component: string) =>
-      getUnderlyingErc20(component, provider)
-    )
-  const wrappedTokens = await Promise.all(underlyingERC20sPromises)
-  const swapData = issuanceComponents.map((_: string, index: number) => {
-    const wrappedToken = wrappedTokens[index]
-    const underlyingERC20 = wrappedToken.underlyingErc20
-    return {
-      dexData: getStaticIssuanceSwapData(inputToken, underlyingERC20.address),
-    }
-  })
-  return swapData
-}
-
-export async function getRedemptionERC4626SwapData(
-  indexTokenSymbol: string,
-  indexToken: string,
-  outputToken: string,
-  indexTokenAmount: BigNumber,
-  provider: JsonRpcProvider
-): Promise<ComponentSwapData[]> {
-  const issuanceModule = getIssuanceModule(indexTokenSymbol)
-  const issuance = new Contract(issuanceModule.address, IssuanceAbi, provider)
-  const [issuanceComponents] =
-    await issuance.getRequiredComponentRedemptionUnits(
-      indexToken,
-      indexTokenAmount
-    )
-  const underlyingERC20sPromises: Promise<WrappedToken>[] =
-    issuanceComponents.map((component: string) =>
-      getUnderlyingErc20(component, provider)
-    )
-  const wrappedTokens = await Promise.all(underlyingERC20sPromises)
-
-  const swapData = issuanceComponents.map((_: string, index: number) => {
-    const wrappedToken = wrappedTokens[index]
-    const underlyingERC20 = wrappedToken.underlyingErc20
-    return {
-      dexData: getStaticRedemptionSwapData(
-        underlyingERC20.address,
-        outputToken
-      ),
```

Comment:
Deleted those and used the `getIssuanceComponentSwapData` as they were already under test in componentSwapData.test.ts but basically very similar. Let me know if I missed anything here tho that was different.